types/react-bootstrap-table-next/index.d.ts

```diff
@@ -512,15 +516,15 @@ export interface ExpandRowProps<T> {
     renderer: (row: T, rowIndex: number) => JSX.Element;
     expanded?: number[];
```

Comment:
@gosticks  Noticed another discrepancy, this shouldn't be `number[]` but rather an Array of the type that is defined as the `keyField`:

https://react-bootstrap-table.github.io/react-bootstrap-table2/docs/row-expand-props.html#expandrowexpanded-array

I also initially thought this might have been the row index, but after checking the storybook it became clearer that this is referring to the value in the `keyField`:

https://react-bootstrap-table.github.io/react-bootstrap-table2/storybook/index.html?selectedKind=Row%20Expand&selectedStory=Expand%20Management&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel

Reply:
I just checked the source and you are right it is using the `dataKey` which actually makes sense since row selection comparably. We should definitely fix that. 
I am a little unsure on what type to use. The issue agains using `Array<T[keyof T]>` is that the library supports multi level key names e.g. `product.info.qty`. Therefore we would need to use `any[]`. This also means I should refactor all uses of T[keyof T] since it technically restricts the use of multi level keys.

Reply:
@gosticks yep, agreed about the complications around supporting the multi-level key names. I updated this to be `any[]`.